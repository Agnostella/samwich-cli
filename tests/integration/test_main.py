import os
import pathlib
import shutil
from click.testing import CliRunner

from samwich_cli.main import cli


def test_should_match_snapshot_given_lambda_and_layer_build(snapshot):
    # Given
    runner = CliRunner()
    example_app = pathlib.Path.cwd() / "docs" / "examples" / "lambda-and-layer"
    with runner.isolated_filesystem():
        for example in example_app.glob("*"):
            if example.is_file():
                shutil.copy(example, example.name)
            else:
                shutil.copytree(example, example.name)

        # When
        result = runner.invoke(
            cli,
            [
                "--template-file",
                "template.yaml",
                "--requirements",
                "requirements.txt",
                "--debug",
            ],
        )
        build_dir = pathlib.Path.cwd() / ".aws-sam"
        built_artifacts = set()
        for root, dirs, files in os.walk(build_dir):
            # Get relative path for this directory
            rel_path = os.path.relpath(root, build_dir)

            # Add all files from this directory
            for file in files:
                file_rel_path = (
                    os.path.join(rel_path, file) if rel_path != "." else file
                )
                built_artifacts.add(file_rel_path)

    # Then
    assert result.exception is None, result.exc_info
    assert result.exit_code == 0, result.output

    assert built_artifacts == snapshot
