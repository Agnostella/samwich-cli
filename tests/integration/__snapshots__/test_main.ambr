# serializer version: 1
# name: test_should_match_snapshot_given_lambda_and_layer_build
  set({
    'build.toml',
    'build/ApiFunction1/app/lambda_1/__init__.py',
    'build/ApiFunction1/app/lambda_1/main.py',
    'build/ApiFunction1/app/lambda_1/repository.py',
    'build/ApiFunction2/app/lambda_2/__init__.py',
    'build/ApiFunction2/app/lambda_2/main.py',
    'build/ApiFunction2/app/lambda_2/repository.py',
    'build/ApiFunction3/app/lambda_3/__init__.py',
    'build/ApiFunction3/app/lambda_3/main.py',
    'build/ApiFunction3/app/lambda_3/repository.py',
    'build/Layer/python/annotated_types-0.7.0.dist-info/METADATA',
    'build/Layer/python/annotated_types-0.7.0.dist-info/RECORD',
    'build/Layer/python/annotated_types-0.7.0.dist-info/WHEEL',
    'build/Layer/python/annotated_types-0.7.0.dist-info/licenses/LICENSE',
    'build/Layer/python/annotated_types/__init__.py',
    'build/Layer/python/annotated_types/py.typed',
    'build/Layer/python/annotated_types/test_cases.py',
    'build/Layer/python/app_lib/__init__.py',
    'build/Layer/python/app_lib/base_repository.py',
    'build/Layer/python/requirements.txt',
    'build/template.yaml',
  })
# ---
